time.js

```diff
@@ -1,12 +1,169 @@
 const rfcFormatter = time.DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS[xxxx][xxxxx]");
 const targetParse = time.ZonedDateTime.prototype.parse;
+
+/**
+ * When formatter is not supplied, uses an RFC DateTimeFormatter that can parse the format
+ * of java.time.ZonedDateTime.toString()
+ * @param {String} text 
+ * @param {time.DateTimeFormatter} formatter 
+ * @returns {time.ZonedDateTime}
+ * @throws {DateTimeParseException} when the text cannot be parsed
+ */
 time.ZonedDateTime.prototype.parse = function (text, formatter = rfcFormatter) {
     return targetParse(text, formatter);
 }
 
-module.exports = time;
+/**
+ * A passthrough function that can allow users to call `toZonedDateTime` on
+ * almost anything that can be converted to a time.ZonedDateTime without first
+ * checking to see if it's already a time.ZonedDateTime. 
+ * @returns {time.ZonedDateTime} this
+ */
+time.ZonedDateTime.prototype.toZonedDateTime = function () {
+    return this;
+}
+
+/**
+ * Change the date of the passed in dt to match the date. This preserves
+ * the hours and minutes even in cases where dt and date span a DST changeover.
+ * @param {time.ZonedDateTime} dt date time to move the date on
+ * @param {time.ZonedDateTime} date date time to move dt's date to
+ * @returns {time.ZonedDateTime} dt with date's date
+ */
+const changeDate = function (dt, date) {
+    return dt.withYear(date.year())
+             .withMonth(date.month())
+             .withDayOfMonth(date.dayOfMonth());
+}
+
+/**
+ * Moves the date portion of the date time to today, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with today's date
+ */
+time.ZonedDateTime.prototype.toToday = function () {
+    return changeDate(this, time.ZonedDateTime.now());
+}
+
+/**
+ * Moves the date portion of the date time to tomorrow, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with tomorrow's date
+ */
+time.ZonedDateTime.prototype.toTomorrow = function () {
+    return changeDate(this, time.ZonedDateTime.now().plusDays(1));
+}
+
+/**
+ * Moves the date portion of the date time to yesterday, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with yesterday's date
+ */
+time.ZonedDateTime.prototype.toYesterday = function () {
+    return changeDate(this, time.ZonedDateTime.now().minusDays(1));
+}
+
+/**
+ * Compares this ZDT to see if it falls between start and end times,
+ * accounting for times that span midnight. start and end can be any type
+ * that has a `toZonedDateTime()` method but only the time portion is used.
+ * @param {*} start the starting time, anything that has a toZonedDateTime method
+ * @param {*} end the ending time, anything that has a toZonedDateTime method
+ * @returns {Boolean} true if this is between start and end
+ */
+time.ZonedDateTime.prototype.betweenTimes = function(start, end) {
+    start = start.toZonedDateTime();
+    end = end.toZonedDateTime();
+    if(end.isBefore(start)) {
```

Comment:
Because we are only looking at the times and ignoring the dates (see above). 

This logic to to handle cases where one wants to see if a time falls between two other times that may span midnight. When that occurs, the start will be before the end (e.g. if we want a time range between a start of 22:00 and an end of 05:00, assuming the dates are the same (see below) the end is before the start.

I made an assumption that the dates were the same. I probably can't do that. I should either `toToday` those start and ends to normalize the dates, or just extract the times via Joda to do the comparisons.